components/incidents/createIncidents/CreateIncident.tsx

```diff
@@ -0,0 +1,363 @@
+import { Button } from 'baseui/button';
+import React, { useEffect, useState } from 'react';
+import styles from "../../../styles/CreateIncident.module.css";
+import {Spinner, SIZE} from 'baseui/spinner';
+import InputStatus from './InputStatus';
+import IncidentName from './IncidentName';
+import IncidentMessage from './IncidentMessage';
+import ComponentsAffected from './ComponentsAffected';
+import {NEXT_PUBLIC_AUTH_TOKEN} from './../../../constants';
+import { ComponentObject } from './ComponentsAffected';
+import Router from 'next/router'
+import { Block } from 'baseui/block';
+import {
+    SnackbarProvider,
+    useSnackbar,
+    DURATION,
+  } from 'baseui/snackbar';
+
+
+interface STATUSType {
+    [key: string]: number
+  }
+
+interface SendComponentObject{
+    [key: string]: string
+}
+
+export interface SpecialEvent{
+    nativeEvent:{
+        offsetX: number
+    },
+    target: {
+        offsetWidth: number,
+        classList: DOMTokenList
+    },
+    
+}
+
+const  STATUS:STATUSType = {
+    "operational" : 1,
+    "degraded_performance": 2,
+    "partial_outage": 3,
+    "major_outage": 4,
+    "under_maintenance": 5,
+}
+
+const getStatus = (id:number) => {
+    switch(id){
+        case 1: return "operational";
+        case 2: return "degraded_performance"
+        case 3: return "partial_outage"
+        case 4: return "major_outage"
+        case 5: return "under_maintenance"
+       
+    }
+}
+
+const getIncidentStatus = (id:String) => {
+    switch(id){
+        case "Investigating" : return "scheduled"
+        case "Identified" : return "in_progress"
+        case "Monitoring" : return "verifying"
+        case "Resolved" : return "completed"
+    }
+}
+
+interface JSONObject{
+    name:String,
+    id:String,
+    status: string
+}
+
+interface optionType{
+    option: {
+        id: Number;
+    }
+}
+
+let InitialData:(ComponentObject|never)[] = [];
+
+//NOTE : id used in component is not the actual id of the component. Instead use compId for the same.
+
+interface CreateIncidentProps{
+    pageID: String[]
+}
+
+export default function CreateIncident (props:CreateIncidentProps) {
+
+   
+    const [isLoaded, setIsLoaded] = useState<number>(0);
+    const [incidentName , setIncidentName] = useState<string>('');
+    const [incidentStatus, setIncidentStatus] = useState<String>("Investigating");
+    const [incidentMessage, setIncidentMessage] = useState<String>('');
+    const [componentsAffected, setComponentsAffected] = useState<ComponentObject[]>([]);
+    const [isSubmitClicked, setIsSubmitClicked] = useState<boolean>(false);
+    const {enqueue, dequeue} = useSnackbar();
+
+    useEffect(() => {
+        const pageID = props.pageID;
+        pageID.forEach((page:String) => {
```

Comment:
all api data should be fetched together and we should show loader instead of creation form when api is in progress. and avoid multiple dependent useEffect

Reply:
Now Showing only loader when api data fetch is in progress